lib/_emerge/Scheduler.py

```diff
@@ -1804,13 +1807,29 @@ def _schedule_tasks(self):
                 and not self._jobs
                 and not self._task_queues.merge
             ):
-                task = self._merge_wait_queue.popleft()
-                task.scheduler = self._sched_iface
-                self._merge_wait_scheduled.append(task)
-                self._task_queues.merge.add(task)
-                task.addExitListener(self._merge_wait_exit_handler)
-                self._status_display.merges = len(self._task_queues.merge)
-                state_change += 1
+                while self._merge_wait_queue:
+                    # If we added non-system packages to the merge queue in a
+                    # previous iteration of this loop, then for system packages
+                    # we need to come back later when the merge queue is empty.
+                    if (
+                        self._task_queues.merge
+                        and self._merge_wait_queue[0].is_system_pkg
+                    ):
+                        break
```

Comment:
We could possibly search through the queue for non-system packages to merge here, but maybe this is good as-is for our initial implementation since this way it keeps the merge order in the same order that the builds finished.

Reply:
Yeah, added this:
```python
# TODO: Maybe promote non-system packages to the front of the
# queue and process them within the current loop, though that
# causes merge order to differ from the order builds finish.
```